packages/camera/camera/android/src/test/java/io/flutter/plugins/camera/CameraRegionUtilsTest.java

```diff
@@ -247,107 +249,180 @@ public void setUp() {
       assertTrue(CameraRegionUtils.MeteringRectangleFactory.create(0, 0, 10, 10, 1).equals(r));
 
       // Bottom right
-      r = CameraRegionUtils.convertPointToMeteringRectangle(this.mockCameraBoundaries, 1.0, 1.0);
+      r =
+          CameraRegionUtils.convertPointToMeteringRectangle(
+              this.mockCameraBoundaries,
+              1.0,
+              1.0,
+              PlatformChannel.DeviceOrientation.LANDSCAPE_LEFT);
       assertTrue(CameraRegionUtils.MeteringRectangleFactory.create(89, 89, 10, 10, 1).equals(r));
 
       // Top left
-      r = CameraRegionUtils.convertPointToMeteringRectangle(this.mockCameraBoundaries, 0.0, 1.0);
+      r =
+          CameraRegionUtils.convertPointToMeteringRectangle(
+              this.mockCameraBoundaries,
+              0.0,
+              1.0,
+              PlatformChannel.DeviceOrientation.LANDSCAPE_LEFT);
       assertTrue(CameraRegionUtils.MeteringRectangleFactory.create(0, 89, 10, 10, 1).equals(r));
 
       // Top right
-      r = CameraRegionUtils.convertPointToMeteringRectangle(this.mockCameraBoundaries, 1.0, 0.0);
+      r =
+          CameraRegionUtils.convertPointToMeteringRectangle(
+              this.mockCameraBoundaries,
+              1.0,
+              0.0,
+              PlatformChannel.DeviceOrientation.LANDSCAPE_LEFT);
+      assertTrue(CameraRegionUtils.MeteringRectangleFactory.create(89, 0, 10, 10, 1).equals(r));
+    } finally {
+      tearDownTestMeteringRectangleFactory();
+    }
+  }
+
+  @Test()
+  public void
+      convertPointToMeteringRectangle_should_rotate_metering_rectangle_according_to_ui_orientation() {
+    setUpTestMeteringRectangleFactory();
+    try {
+      MeteringRectangle r;
+      // PORTRAIT_UP
+      r =
+          CameraRegionUtils.convertPointToMeteringRectangle(
+              this.mockCameraBoundaries, 1, 1, PlatformChannel.DeviceOrientation.PORTRAIT_UP);
       assertTrue(CameraRegionUtils.MeteringRectangleFactory.create(89, 0, 10, 10, 1).equals(r));
+      // PORTRAIT_DOWN
+      r =
+          CameraRegionUtils.convertPointToMeteringRectangle(
+              this.mockCameraBoundaries, 1, 1, PlatformChannel.DeviceOrientation.PORTRAIT_DOWN);
+      assertTrue(CameraRegionUtils.MeteringRectangleFactory.create(0, 89, 10, 10, 1).equals(r));
+      // PORTRAIT_UP
+      r =
+          CameraRegionUtils.convertPointToMeteringRectangle(
+              this.mockCameraBoundaries, 1, 1, PlatformChannel.DeviceOrientation.LANDSCAPE_RIGHT);
+      assertTrue(CameraRegionUtils.MeteringRectangleFactory.create(0, 0, 10, 10, 1).equals(r));
+      // LANDSCAPE_LEFT (no rotation)
+      r =
+          CameraRegionUtils.convertPointToMeteringRectangle(
+              this.mockCameraBoundaries, 1, 1, PlatformChannel.DeviceOrientation.LANDSCAPE_LEFT);
+      assertTrue(CameraRegionUtils.MeteringRectangleFactory.create(89, 89, 10, 10, 1).equals(r));
+
+    } finally {
+      tearDownTestMeteringRectangleFactory();
```

Comment:
nit: what about moving these to a separate test class with their own setUp/tearDown handlers?

Reply:
Tests split up into a separate file